traitsui/tests/editors/test_image_enum_editor.py

```diff
@@ -0,0 +1,488 @@
+            )
+
+            # Select new image
+            click_on_image(get_button_control(image_grid_control, 1))
+            gui.process_events()
+
+            self.assertEqual(enum_edit.value, 'top right')
+
+            # Check that dialog window is closed
+            self.assertEqual(list(control.GetChildren()), [])
+
+    @skip_if_not_qt4
+    @unittest.skipIf(is_linux, "Issue enthought/traitsui#854")
+    def test_simple_editor_combobox(self):
+        enum_edit = EnumModel()
+
+        with store_exceptions_on_all_threads():
+            gui, combobox = self.setup_gui(enum_edit, get_view("simple"))
+
+            self.assertEqual(enum_edit.value, 'top left')
+
+            # Smoke test for ImageEnumItemDelegate painting
+            combobox.showPopup()
+            gui.process_events()
+
+            combobox.setCurrentIndex(1)
+            combobox.hidePopup()
+            gui.process_events()
+
+            self.assertEqual(enum_edit.value, 'top right')
+
+
+@skip_if_null
+class TestCustomImageEnumEditor(unittest.TestCase):
+
+    def setup_gui(self, model, view):
+        gui = GUI()
+        ui = model.edit_traits(view=view)
+        self.addCleanup(ui.dispose)
+
+        gui.process_events()
+        editor = ui.get_editors("value")[0]
+        control = editor.control
+
+        return gui, control
+
+    def test_custom_editor_more_cols(self):
+        # Smoke test for setting up an editor with more than one column
+        enum_edit = EnumModel()
+        view = View(
+            UItem(
+                'value',
+                editor=ImageEnumEditor(
+                    values=[
+                        'top left', 'top right', 'bottom left', 'bottom right'
+                    ],
+                    prefix='@icons:',
+                    suffix='_origin',
+                    path='dummy_path',
+                    cols=4,
+                ),
+                style="custom",
+            ),
+            resizable=True,
+        )
+
+        with store_exceptions_on_all_threads():
+            self.setup_gui(enum_edit, view)
+
+    def test_custom_editor_selection(self):
+        enum_edit = EnumModel()
+
+        with store_exceptions_on_all_threads():
+            gui, control = self.setup_gui(enum_edit, get_view("custom"))
+            self.assertEqual(
+                get_button_strings(control),
+                ['top left', 'top right', 'bottom left', 'bottom right']
+            )
+
+            self.assertEqual(enum_edit.value, 'top left')
+            self.assertEqual(
+                get_all_button_selected_status(control),
+                [True, False, False, False]
+            )
+
+            click_on_image(get_button_control(control, 1))
+            gui.process_events()
+
+            self.assertEqual(enum_edit.value, 'top right')
+            self.assertEqual(
+                get_all_button_selected_status(control),
+                [False, True, False, False]
+            )
```

Comment:
The assertion on the button status is actually testing Qt/Wx, rather than testing Traitsui editor, because the change is driven by clicking the widget.
The assertion would be meaningful if the change is caused by, e.g. an assignment on `enum_edit.value`, like the test below. Could you remove the button status assertion in this test please?